src/Properties/ModelPropertyExtension.php

```diff
@@ -232,12 +232,21 @@ private function castPropertiesType(Model $modelInstance): void
                 continue;
             }
 
+            // Reduce encrypted castable types
+            if (in_array($type, ['encrypted', 'encrypted:array', 'encrypted:collection', 'encrypted:json', 'encrypted:object'], true)) {
+                $type = Str::after($type, 'encrypted:');
+            }
+
+            // Truncate cast parameters
+            $type = Str::before($type, ':');
```

Comment:
Looking through the framework, it appears to be explicitly looking for `decimal:`, `date:`, `datetime:` etc for the built-in parameterized casts. That said, I think for the interest maintainibility here, we can reasonably assume that everything following the `:` is parameters and not part of the underlying cast type.